integration-test.py

```diff
@@ -503,9 +503,17 @@ def query_result(self, query):
                          params=dict(data=query))
             if r.status_code == 200:
                 break
-            if r.status_code != 429:
+            if r.status_code != 429 and r.status_code != 504:
```

Comment:
I like `r.status_code not in (429, 504)`  - much better than testing 429, 504 separately.